### DISCUSSION TITLE
New Ride Type: Wing Coaster

### DISCUSSION THREAD

**Discussion by karst** [OpenRCT2/OpenRCT2#21012] [Feature requests]
Inspired by B&M's many wing coasters, this should become a new ride type with a new ride vehicle. 
I have not been able to get in contact with B&M as of yet. If we cannot get permission a inspired vehicle should be in the game instead.

###### Concept using custom vehicle:
![image](https://github.com/OpenRCT2/OpenRCT2/assets/1516893/004dffcc-0b33-4bf3-a912-5ddd9069c86e)

What needs to be done to create the wing coaster ride type:

- [ ] 3 tile wide stations.
This will allow for the vehicle to be in the station without clipping

- [ ] An entrance/exit on each side of the new station required.
This is so people can load in on only the side they queued for. Perhaps the banners could also say what side (right/left in brackets)

- [ ] Official support of queue splitting.
Currently queue splitting is only supported visually using the tile inspector.

- [ ] New track pieces
The zero-g roll, large corkscrew and large zero-g roll should be included with this new ride type at least. Sprites for this have been rendered, they just need to be edited by hand.

What differentiates the wing coaster from the twister.

- Slightly higher intensity rating, but also exciting rating.
- Launched shuttle mode
- Rveverse start launched shuttle mode
- Contious circuit mode/block sectioned mode
- 3 wide station only, instead of regular station.
- Has access to boosters, inversions and lifthills.
- Near miss bonus (?)

If you have more ideas I'd love to hear it.
I want this ride type in the pipeline, but it has no rush.

**Comment by ocalhoun6** [comment]
- Should probably maintain the 3-wide clearance requirement for the entire track. (Though that would mean that certain existing track elements like the small vertical loop would have to be redesigned or omitted.)
- Would be cool if it had wide track for lift hills [like this](https://www.coaster101.com/wp-content/uploads/2012/05/DSC00606.jpg).

**Reply by karst** [reply]
Small vertical loop should probably be omitted from this ride type yes. I think it's a good idea for the clearance. Does mean it would need sprites for the tunnels

**Reply by karst** [reply]
I would love to see the introduction of catwalks. But i do think this should be a universal addition, a scenery element you can plop around a piece of track similar how benches, lamps, jumping fountains, bins and such work on paths.

**Comment by kyphii** [comment]
Extra-wide stations (or any track for that matter) sounds like it would be a coding nightmare - why not instead use an alternate station platform style, like the multi-dim does? 

![image](https://github.com/OpenRCT2/OpenRCT2/assets/8711258/3fe84986-9194-4167-9101-0f0a6bcb78c6)
(Messy example but you get the idea)

Could probably slim the wing coaster CTR sprites down by a few pixels as well - I doubt most players would notice or care if they were a bit smaller than would be realistic (The game's scale is already bonkers anyways).

**Reply by karst** [reply]
I'd honestly like to see the same thing implemented as above on the multi-dimension. The station would be a 3x1 working from the centre. Would have the same vehicle location drawing as any straight piece would, just more clearance. I do think that making it check on the sides of the entire track would become a coding nightmare so definitely overkill there.

**Comment by TheClaudeQC** [comment]
I have already an idea for a possible custom variant for this one. I just need to know what I need to create the trains?

**Comment by X123M3-256** [comment]
> why not instead use an alternate station platform style, like the multi-dim does?

Because wing coaster cars would be much wider - they significantly exceed the width of the tile, so no matter how the station platform is drawn, if it's only one tile the seats would be hanging off the edge and clipping through the station buildings. A similar issue exists with tunnels. Multi dimension cars are actually also wider than a tile but only slightly so the clipping isn't that noticeable.

Bear in mind that my old wing coaster CTR shown in these screenshots was rendered too small - if it was drawn the same scale as the hybrid and single rail trains it would be about 10% wider still than is shown in those images.
